routers/main.js

```diff
@@ -44,32 +44,45 @@ router.get('/createSuggestion',
 		req.session.title = req.query.title;
 		req.session.description = req.query.description;
 		req.session.state = req.query.state;
+		req.session.originalUrl = '/createSuggestion';
 		next();
 	},
 	passport.authenticate('github', {scope: [ 'repo' ]}));
 
 router.get('/github_callback', passport.authenticate('github', {failureRedirect: '/authKO'}), (req, res) => {
+
 	const suggestionsController = new SuggestionsController(config);
-	suggestionsController.createPullRequest(process.env.npm_config_ludwig_accessToken, req.session.title, req.session.description, req.session.state, res, config.github.branch);
+	if (req.session.originalUrl === '/createSuggestion') {
+		suggestionsController.createPullRequest(process.env.npm_config_ludwig_accessToken, req.session.title, req.session.description, req.session.state, res, config.github.branch);
+	} else {
+		res.redirect(req.session.originalUrl);
+	}
 });
 
+router.get('/listTestsConnected', (req, res, next) => {
+	req.session.originalUrl = '/listTests?filter=mine';
+	if (req.session.passport) {
+		res.redirect('/listTests?filter=mine');
```

Comment:
The aim of this bit is to not go through the authentication process if the user already has session data present (found it somewhat slow and could not put my finger on why it was that way). We could enforce authentication each time, but, as this endpoint is just there to filter tests based on the username present in the session, I figured it'd avoid calls to GH and provide an acceptable result. Maybe we could just as well enforce authentication each time.